addons/portal/wizard/portal_share.py

```diff
@@ -84,6 +94,20 @@ def action_send_mail(self):
                 partner_ids=[(6, 0, partner.ids)])
             self = self.with_context(lang=saved_lang)
 
+    def action_send_mail(self):
+        active_record = self.env[self.res_model].browse(self.res_id)
+        note = self._get_note()
+        signup_enabled = self.env['ir.config_parameter'].sudo().get_param('auth_signup.invitation_scope') == 'b2c'
+
+        if hasattr(active_record, 'access_token') and active_record.access_token or not signup_enabled:
+            partner_ids = self.partner_ids
+        else:
+            partner_ids = self.partner_ids.filtered(lambda x: x.user_ids)
+        # if partner already user or record has access token send common link in batch to all user
```

Comment:
There must be a subtetly I'm missing but `_send_public_link` seems to be sending the common link to everyone, regardless of their status.

Reply:
Before my change, it is true, you miss nothing. In fact, if we are in the case where `not active_record.access_token and signup_enabled` then we will send 2 times the same mail if I correctly understood the initial code.
That's why I change the code to send once the mail for each partner in both cases.
I will test it to be sure before "rebasing" my branch. 🙂

Reply:
Oups sorry, my previous comment was wrong. I called this method like this because IMO, it is a "public link" sent to everyone.
Perhaps, you prefer rename it to `_send_share_link` ? 🤔

Reply:
I don't necessarily mind either way, just pointing out that the comment doesn't seem to match the behaviour.